Issue 42270: Promo figures for Panorama Public datasets

vagisha opened the pull request:
#### Rationale
We want to request submitters to give us a short text and an image describing their Panorama Public dataset when they make it public. This will be included in the slideshow on the PanoramaWeb homepage. 

#### Changes
- After making their data public, submitters will see a button they can click to provide an entry (description + image) for the slideshow catalog.
- Enable cropping the selected image to required aspect ratio in the catalog entry form before uploading to the server.
- A CatalogEntryWebPart is added to the experiment details page.  This webpart, visible only to folder admins and Panorama Public submitters, will display the entry.  Or, if an entry does not exist for a public dataset, the webpart will display a button to add an entry.
- A Panorama Public admin can review the entry and approve or reject in the CatalogEntryWebPart or in the CatalogEntryTableInfo query.
- The Panorama Public admin console has a link for "Panorama Public Catalog Settings" that can be used to configure the max image file size, image dimensions, character limit for the description, and the maximum entries to display in the slideshow.
- GetCatalogApiAction action can be used for getting a list of approved entries in JSON format. A test page (viewSlideshow.jsp) has been added as an example use case.

vagisha committed:
Issue 42270: Promo figures for Panorama Public datasets
- After making their data public, submitters will see a button they can click to provide an entry for thier data in the slideshow catalog.
- Enable cropping image to desired aspect ratio in the brower before uploading to the server
- A CatalogEntryWebPart is added to the experiment details page that can be used to add an entry for existing public datasets. The webpart will display entry details if it already exists for the experiment.
- A Panorama Public admin can review the entry and approve or reject in the CatalogEntryWebPart or the CatalogEntryTableInfo query.
- The Panorama Public admin console has a link for "Panorama Public Catalog Settings" that can be used to configure the max image file size, image dimensions, character limit for the description, and the maximum entries to display in the slideshow.
- GetCatalogApiAction action can be used for getting a list of approved entries in JSON format. An test page (viewSlideshow.jsp) has been added as an example use case.

vagisha committed:
If the number of entries to be displayed in the slideshow is less than the total number of approved entries, the date is used to index into the set of entries so that all entries will be displayed in "total approved entries" / "entries to display" days.

vagisha committed:
Fix condition for getting an entry sublist.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/query/CatalogEntryTableInfo.java:
```diff
@@ -0,0 +1,196 @@
+            public void addQueryFieldKeys(Set<FieldKey> keys)
+            {
+                super.addQueryFieldKeys(keys);
+                keys.add(FieldKey.fromParts("Id"));
```
I'm not seeing where `Id` is being used. If it's needed, it's best practice to prefix it with the current column's FieldKey's parent too, as that will make it more likely to work if the table has been included in a custom query, as a lookup, etc.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/query/CatalogEntryTableInfo.java:
```diff
@@ -0,0 +1,196 @@
+            public void addQueryFieldKeys(Set<FieldKey> keys)
+            {
+                super.addQueryFieldKeys(keys);
+                keys.add(FieldKey.fromParts("ShortUrl"));
```
Similar advice here - include the current column's FieldKey parent both here and when you pull it during `renderGridCellContents()`

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/view/publish/catalogEntryForm.jsp:
```diff
@@ -0,0 +1,371 @@
+        </table>
+        <br>
+        <%=button("Submit").submit(true).disableOnClick(true)%>
+        &nbsp
```
```suggestion
        &nbsp;
```

vagisha committed:
- Code review: include current column's FieldKey parent.
- Add "View Entry" button in success view after saving an entry.
- Add "Back to Folder" button in CatalogEntryWebPart when viewing entry details.

vagisha committed:
Merge branch 'release22.11-SNAPSHOT' into 22.11_fb_42270_panorama-public-slideshow

vagisha committed:
Fix compilation error after merging release22.11-SNAPSHOT
